lib/vehicle_factory.rb

```diff
@@ -0,0 +1,28 @@
+class VehicleFactory
+  attr_reader :created_vehicles 
+  
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles(vehicle_details)
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    wa_registrations(vehicle_details)
+    @created_vehicles
+  end 
+  
+   # Helper method
+  def wa_registrations(vehicle_details)
```

Comment:
This is not tested currently. Think about other enumerables that return an array of the same length.